packages/fxa-auth-server/lib/routes/devices-and-sessions.js

```diff
@@ -475,29 +407,40 @@ module.exports = (log, db, config, customs, push, pushbox, devices, oauthdb) =>
 
         const credentials = request.auth.credentials;
 
+        // XXX: Ideally, we would call oauthdb.listAuthorizedClients here, and perform a similar merging
+        // to what we do in /account/attached_clients. That's awkward to do because this route can be called
+        // with a refreshToken, but oauthdb currently requires a sessionToken. Let's defer that to followup.
+
         return request.app.devices
           .then(deviceArray => {
             return deviceArray.map(device => {
-              return Object.assign({
+              const formattedDevice = {
                 id: device.id,
-                isCurrentDevice: !! ((credentials.id && credentials.id === device.sessionToken) ||
+                isCurrentDevice: !! ((credentials.id && credentials.id === device.sessionTokenId) ||
                   (credentials.refreshTokenId && credentials.refreshTokenId === device.refreshTokenId)),
-                location: marshallLocation(device.location, request),
+                lastAccessTime: device.lastAccessTime,
+                location: device.location,
```

Comment:
@rfk, I believe this change (the removal of `marshallLocation`) is the cause of:

https://sentry.prod.mozaws.net/operations/auth-prod/issues/6027903/?environment=prod

The response schema doesn't include `countryCode` in `location` and `marshallLocation` was responsible for stripping it out. What was the reason for getting rid of it here?

Reply:
Ah right, just seen noticed it moved down into `lib/routes/utils/clients.js`, but now it falls back to leaving the unadulterated `location` property behind instead of an empty object. I can fix it now, you can ignore the above sorry.

Reply:
Ref: https://github.com/mozilla/fxa/pull/1699

Reply:
Ugh, sorry :-(

> What was the reason for getting rid of it here?

There's no particular unifying vision behind changing this specifically, it just got swept up in a broader refactor to share code between these routes.